tools/pre-commit

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+#
+# Copyright 2025 SmartThings, Inc.
+# Licensed under the Apache License, Version 2.0
+#
+# pre-commit
+#
+# - Copy to .git/hooks/pre-commit
+# - Ensure executable
```

Comment:
I would suggest just making sure this is committed as executable to avoid the hassle.

Also I would suggest to use a softlink instead of copying, so that you don't need to re-copy it later (`ln -s ../../tools/pre-commit .git/hooks/pre-commit` (use a relative link so you can move the directory later)).